TempDirectory.Test/TempDirectoryBuilderTest.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using System.IO;
+using Messerli.TempDirectory;
+using Xunit;
+using JetBrains.Annotations;
+
+namespace TempDirectory.Test
+{
+    public class TempDirectoryBuilderTest
+    {
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void CreatesTempDirectory(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.True(Directory.Exists(tempDirectory.FullName));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void DeletesTempDirectory(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            var tempDirectory = tempDirectoryBuilder.Create();
+            Assert.True(Directory.Exists(tempDirectory.FullName));
+            tempDirectory.Dispose();
+
+            Assert.False(Directory.Exists(tempDirectory.FullName));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void FullNameContainsName(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.Contains(tempDirectory.Name, tempDirectory.FullName);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void FullNameIsNotName(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.NotEqual(tempDirectory.FullName, tempDirectory.Name);
+            }
+        }
+
+        [Fact]
+        public void PrefixIsInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder().Prefix(Prefix).Create())
+            {
+                Assert.Contains(Prefix, tempDirectory.Name);
```

Comment:
Test that the name starts with the prefix

Reply:
Addressed in https://github.com/messerli-informatik-ag/temp-directory/pull/1/commits/6793941234fe7ebc0904638a7c90d0a31a54c7be